software/contrib/consequencer.py

```diff
@@ -255,209 +273,391 @@ class pattern:
     SN=[]
     HH=[]
 
+    # Initialize pattern probabilities
+
+    BdProb = []
+    SnProb = []
+    HhProb = []
+
+    # Mixed probability patterns
+    BD.append("10111111111100001011000000110000")
+    SN.append("10001000100010001010000001001000")
+    HH.append("10101010101010101010101010101010")
+    BdProb.append("99992111129999999999999999969999")
+    SnProb.append("99999999999999999999999999999999")
+    HhProb.append("92939495969792939495969792939492")
+
+    BD.append("10111111111100001011000000110000")
+    SN.append("10001000100010001010000001001000")
+    HH.append("11111111111111111111111111111111")
+    BdProb.append("99992222229999999999999999999999")
+    SnProb.append("99999999999999999999999999999999")
+    HhProb.append("44449999555599996666999922229999")
+
+    BD.append("1000100010001000")
+    SN.append("0000101001001000")
+    HH.append("0101010101010101")
+    BdProb.append("9999999999999999")
+    SnProb.append("9999999999999999")
+    HhProb.append("9999999999999999")
+
+    BD.append("1000110010001100")
+    SN.append("0000101001001000")
+    HH.append("1111111111111111")
+    BdProb.append("9999939999999299")
+    SnProb.append("9999999999999999")
+    HhProb.append("9293949592939495")
+
     # African Patterns
     BD.append("10110000001100001011000000110000")
     SN.append("10001000100010001010100001001010")
     HH.append("00001011000010110000101100001011")
+    BdProb.append("99999999999999999999999999999999")
```

Comment:
[optional] Since the default is to have all 9's for each step of each pattern, it's possible to reduce the number of lines added to this script by dynamically setting up probability. Something like this, after all patters are defined:
```python
for idx in range(len(BD)):
  BdProb.append("9" * len(BD[idx]))
  SnProb.append("9" * len(SD[idx]))
  HhProb.append("9" * len(HH[idx]))
```
For example:
```python
>>> b = ['aaa','bbbbb','ccc']
>>> s = ['xxx','yyyyy','zzz']
>>> bp = []
>>> sp = []
>>> for idx in range(len(b)):
...    bp.append("9" * len(b[idx]))
...    sp.append("9" * len(s[idx]))
... 
>>> bp
['999', '99999', '999']
>>> sp
['999', '99999', '999']
```

Reply:
It might be simpler to have a special case value indicate that the pattern always has 100% probability. Something that isn’t a list that would be easy to check for. None or maybe int 100.

Reply:
Thank you both for your comments. You both inspired me to tweak this. The new commit permits the entry of a single vale e.g. '9' as a probability, if this is detected when loaded, it is automatically filled with 9s in the probability pattern that will be used.